src/store/TriggerLocalStore.ts

```diff
@@ -0,0 +1,25 @@
+import { useRef } from "react";
+import { create} from 'zustand';
+import {devtools} from 'zustand/middleware';
+import {shallow} from 'zustand/shallow';
+import createTriggersSlice, {TriggersSlice} from './TriggersSlice';
+
+const useTriggerLocalStore = () => {
+    const useStoreRef = useRef(
+        create<TriggersSlice>()(
+            devtools(
+              (...a) => ({
+                ...createTriggersSlice(...a),
+              }),
+              {name: 'Local Store Zustand', enabled: process.env.NODE_ENV === 'development'}
+            )
+          )
+    );
+    const useStore = useStoreRef.current;
+          
+    const useShallowLocalStore = <U>(selector: (state: TriggersSlice) => U) => useStore(selector, shallow);
+
+    return [useShallowLocalStore];
+}; 
+
+export default useTriggerLocalStore;
```

Comment:
this is a draft implementation of local store.
Usage would be:

`  const [useShallowLocalStore] = useTriggerLocalStore();
// whatever props we want to fetch from store
  const {text, setText} = useShallowLocalStore(state => ({
    text: state.text,
    setText: state.setText,
  }));`

Reply:
Tested it locally with 2 instances of the same store on different components

Reply:
Nice one! Is there a way to provide the initial state?

Reply:
I believe it is provided in TriggersSlice itself, isn't is @haneabogdan ?